src/NUnitFramework/framework/Api/NUnitTestAssemblyRunner.cs

```diff
@@ -245,17 +245,37 @@ public bool WaitForCompletion(int timeout)
         }
 
         /// <summary>
+        /// Signal any test run that is in process to stop. Return without error if no test is running.
+        /// </summary>
+        /// <param name="force">If true, kill any tests that are currently running</param>
+        public void StopRun(bool force)
+        {
+            if (IsTestRunning)
+            {
+                Context.ExecutionStatus = force
+                    ? TestExecutionStatus.AbortRequested
+                    : TestExecutionStatus.StopRequested;
+
+                Context.Dispatcher.CancelRun(force);
+            }
+        }
+
+        #endregion
+
+        #region Helper Methods
+
+        /// <summary>
         /// Initiate the test run.
         /// </summary>
-        public void StartRun(ITestListener listener)
+        private void StartRun(ITestListener listener)
```

Comment:
Minor organization, but I think that StartRun should come before StopRun and be in the same region now that they are both public. Does this change our internal API at all making this public?